packages/compiler-vapor/__tests__/transforms/transformElement.spec.ts

```diff
@@ -1,2 +1,142 @@
-// TODO: add tests for this transform
-test('baisc', () => {})
+import { makeCompile } from './_utils'
+import {
+  IRNodeTypes,
+  transformElement,
+  transformVBind,
+  transformVOn,
+} from '../../src'
+import { NodeTypes } from '@vue/compiler-core'
+
+const compileWithElementTransform = makeCompile({
+  nodeTransforms: [transformElement],
+  directiveTransforms: {
+    bind: transformVBind,
+    on: transformVOn,
+  },
+})
+
+describe('compiler: element transform', () => {
+  test.todo('baisc')
+
+  // This v-on feature comes from the unit tests of compiler-core,
+  // but I am not sure if Vapor needs this feature.
+  test.fails('props merging: event handlers', () => {
+    const { code, ir } = compileWithElementTransform(
+      `<div @click.foo="a" @click.bar="b" />`,
+    )
+    expect(code).toMatchSnapshot()
+
+    expect(ir.operation).toMatchObject([
+      {
+        type: IRNodeTypes.SET_EVENT,
+        element: 1,
+        key: {
+          type: NodeTypes.SIMPLE_EXPRESSION,
+          content: 'click',
+          isStatic: true,
+        },
+        value: [
+          {
+            type: NodeTypes.SIMPLE_EXPRESSION,
+            content: `a`,
+            isStatic: false,
+          },
+          {
+            type: NodeTypes.SIMPLE_EXPRESSION,
+            content: `b`,
+            isStatic: false,
+          },
+        ],
+      },
+    ])
+  })
```

Comment:
This v-on feature comes from the unit tests of `compiler-core`, but I am not sure if Vapor needs this feature. because the code generated now seems to work very well with multiple `EventListeners` registered.

If we are sure we need it, perhaps we can implement this feature in another PR since it should be unrelated to `setProp`. If this feature is not needed, I can remove this unit test.

Reply:
It's nice to have. It's worth merging multiple event registers into a single one. I modified this unit test, see the latest IR structure.